tools/scalajsenv.js

```diff
@@ -144,13 +144,28 @@ const $idHashCodeMap = ScalaJS.g["WeakMap"] ? new ScalaJS.g["WeakMap"]() : null;
 //!endif
 //!endif
 
+ScalaJS.ClassData = {};
+
 // Core mechanism
 
-ScalaJS.makeIsArrayOfPrimitive = function(primitiveData) {
+ScalaJS.expandSubtypeArray = function(tags){
+  let expanded = [];
+  let len = 0;
+  for (let i = 0; i < tags.length; i++) {
+    let interval = tags[i];
+    let start = interval[0];
+    let end = interval[1];
```

Comment:
You can win a bit of space (and possibly memory usage) by flattening `tags`. Instead of having
```js
tags = [[3, 6], [15, 17]]
```
you could have
```js
tags = [3, 6, 15, 17]
```
Less brackets. Then the loop becomes:
```js
for (let i = 0; i < tags.length; i += 2) {
  const start = tags[i];
  const end = tags[i + 1];
  ...
}
```
(btw, use `const`s where possible)